internal/lookout/repository/query.go

```diff
@@ -50,3 +57,317 @@ func (r *SQLJobRepository) GetQueueStats() ([]*lookout.QueueInfo, error) {
 	}
 	return result, nil
 }
+
+func (r *SQLJobRepository) GetJobsInQueue(opts *lookout.GetJobsInQueueRequest) ([]*lookout.JobInfo, error) {
+	rows, err := r.queryJobsInQueue(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	result := parseJobsInQueueRows(rows)
+	return result, nil
+}
+
+type jobsInQueueRow struct {
+	JobId     string
+	Queue     string
+	Owner     string
+	JobSet    string
+	Priority  sql.NullFloat64
+	Submitted pq.NullTime
+	Cancelled pq.NullTime
+	JobJson   sql.NullString
+	RunId     sql.NullString
+	Cluster   sql.NullString
+	Node      sql.NullString
+	Created   pq.NullTime
+	Started   pq.NullTime
+	Finished  pq.NullTime
+	Succeeded sql.NullBool
+	Error     sql.NullString
+}
+
+func (r *SQLJobRepository) queryJobsInQueue(opts *lookout.GetJobsInQueueRequest) ([]*jobsInQueueRow, error) {
+	query := makeGetJobsInQueueQuery(opts)
+	rows, err := r.db.Query(query)
+	if err != nil {
+		return nil, err
+	}
+
+	joinedRows := make([]*jobsInQueueRow, 0)
+	for rows.Next() {
+		row := &jobsInQueueRow{}
+		err := rows.Scan(
+			&row.JobId,
+			&row.Owner,
+			&row.JobSet,
+			&row.Priority,
+			&row.Submitted,
+			&row.Cancelled,
+			&row.JobJson,
+			&row.RunId,
+			&row.Cluster,
+			&row.Node,
+			&row.Created,
+			&row.Started,
+			&row.Finished,
+			&row.Succeeded,
+			&row.Error,
+		)
+		if err != nil {
+			return nil, err
+		}
+		joinedRows = append(joinedRows, row)
+	}
+	return joinedRows, nil
+}
+
+func makeGetJobsInQueueQuery(opts *lookout.GetJobsInQueueRequest) string {
+	query := `
+		SELECT job.job_id as job_id,
+			   job.owner as owner,
+			   job.jobset as jobset,
+               job.priority as priority,
+               job.submitted as submitted,
+               job.cancelled as cancelled,
+			   job.job as job_json,
+			   job_run.run_id as run_id,
+			   job_run.cluster as cluster,
+			   job_run.node as node,
+			   job_run.created as created,
+			   job_run.started as started,
+			   job_run.finished as finished,
+			   job_run.succeeded as succeeded,
+			   job_run.error as error
+		FROM job LEFT JOIN job_run ON job.job_id = job_run.job_id
+		WHERE job.job_id IN (%s)`
```

Comment:
Does this open us up to SQL injection? What happens if I request a job ID of `3); DROP TABLE job; --`?